src/editor.js

```diff
@@ -722,98 +724,56 @@ export class OdooEditor extends EventTarget {
-            sel.anchorNode === range.startContainer && sel.anchorOffset === range.startOffset;
-        let pos1 = [range.startContainer, range.startOffset];
-        let pos2 = [range.endContainer, range.endOffset];
-        // A selection spanning multiple nodes and ending at position 0 of a
-        // node, like the one resulting from a triple click, are corrected so
-        // that it ends at the last position of the previous node instead.
-        if (!pos2[1]) {
-            pos2 = rightPos(leftDeepOnlyPath(...pos2).next().value);
-            const previousElement = leftDeepOnlyPath(...pos2).next().value;
-            pos2[0] = latestChild(previousElement);
-            pos2[1] = nodeSize(pos2[0]);
-        }
-
-        // Hack: we will follow the logic "do many backspace until the
-        // selection is collapsed". The problem is that after one backspace
-        // the node the starting position of the cursor is in may have
-        // changed (split, removed, moved, etc) so there is no reliable
-        // way to know when there has been enough backspaces... except for
-        // this hack: we put a fake element acting as a one-space-to-remove
-        // element (like an image) at the selection start position and we
-        // hit backspace until that element is removed.
-        if (pos1[0].nodeType === Node.TEXT_NODE) {
-            // First, if the selection start is in a text node, we have to
-            // split that text node to be able to put the fake element
-            // in-between.
-            const splitNode = pos1[0];
-            const splitOffset = pos1[1];
-            const willActuallySplitPos2 =
-                pos2[0] === splitNode && splitOffset > 0 && splitOffset < splitNode.length;
-            pos1 = [splitNode.parentNode, splitTextNode(splitNode, splitOffset)];
-            if (willActuallySplitPos2) {
-                if (pos2[1] < splitOffset) {
-                    pos2[0] = splitNode.previousSibling;
+        let range = getDeepRange(this.document, { sel, splitText: true, select: true });
+        if (!range) return;
+        // "Correct" triple click.
+        if (!range.endOffset && !range.endContainer.previousSibling) {
+            const previous = previousLeaf(range.endContainer, this.dom);
+            previous && range.setEndAfter(previous);
+        }
+        const start = range.startContainer;
+        let end = range.endContainer;
+        // Let the DOM split and delete the range.
+        const doJoin = closestBlock(start) !== closestBlock(range.commonAncestorContainer);
+        let next = nextLeaf(end, this.dom);
+        range.deleteContents();
+        setCursor(start, nodeSize(start));
+        range = getDeepRange(this.document, { sel });
+        // If the end container was fully selected, extractContents may have
+        // emptied it without removing it. Ensure it's gone.
+        while (end && end !== this.dom && !end.contains(range.endContainer) && !isVisible(end, 1)) {
```

Comment:
Hey, better golfing idea. Just set `true` as the default value of the second parameter of `isVisible` ! Then no more magic `1` and you gain 3 chars, woot ! ⛳

Reply:
I could rename `isVisible` into `iV` if you prefer.